server.go

```diff
@@ -1944,6 +1953,33 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 	return nil
 }
 
+// SetSendCompressor sets the compressor that will be used when sending
+// RPC payload back to the client. It may be called at most once, and must not
+// be called after any event that causes headers to be sent (see SetHeader for
+// a complete list). Provided compressor is used when below conditions are met:
+//
+//   - compressor is registered via encoding.RegisterCompressor
+//   - compressor name exists in the client advertised compressor names sent in
+//     grpc-accept-encoding metadata.
+//
+// The context provided must be the context passed to the server's handler.
+//
+// The error returned is compatible with the status package.  However, the
+// status code will often not match the RPC status as seen by the client
+// application, and therefore, should not be relied upon for this purpose.
+func SetSendCompressor(ctx context.Context, name string) error {
+	stream, ok := ServerTransportStreamFromContext(ctx).(*transport.Stream)
+	if !ok || stream == nil {
+		return status.Errorf(codes.Internal, "grpc: failed to fetch the stream from the context %v", ctx)
+	}
+
+	if err := grpcutil.ValidateSendCompressor(name, stream.ClientAdvertisedCompressors()); err != nil {
```

Comment:
I felt this method would be better suited in `internal/grpcutil/compressor.go` (I added it earlier for compressor utilities). This is exposed internally only which should be okay right?

Reply:
If my memory serves me right, things were put into `internal/grpcutil/compressor.go` to ensure that we don't export more APIs than what is actually required from the `encoding` package. But, the functions added as part of this PR to `internal/grpcutil/compressor.go` are only used from here. So, I think it makes more sense to put them here instead of there. What do you think?

I agree that putting them in `internal/grpcutil/compressor.go` will not expose them, but I feel it makes more sense for them to be here, where they are actually used.

Reply:
Makes sense, moved it into the same package